### PR TITLE
[TS] LPS-157425

### PR DESCRIPTION
**Pull request by ericyanLr**
https://issues.liferay.com/browse/LPS-157425

Hi Team,

**Context**
This PR was created with guidance from [PTR-3915](https://issues.liferay.com/browse/PTR-3915) and the plan is for the Search Team and Headless Team to review.

This is a follow-up PR to: https://github.com/liferay-search/liferay-portal/pull/953

**Summary of Changes**
- Search
  - Update mapping to introduce multi-field keyword fields to replace the previous **\*_\<languageId>*_sortable** fields
    - As discussed, we will proceed with adding the sub-field: **\*_\<languageId>*_sortable.keyword_lowercase**
  - Also added mappings for Solr
- Headless API
  - For fields that are utilizing **\*_\<languageId>*_sortable** fields for filtering String fields
    - Use the associated sub-field: **\*_\<languageId>*_sortable.keyword_lowercase**
  - Added tests by updating the template: **base_resource_test_case.ftl** in portal-tools-rest-builder
    - Regenerated relevant tests by running gw buildRest

Please let me know if you have any questions.
Thanks!

### CODE REVIEW

**Comment by ericyanLr** [comment]
ci:test:relevant

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/resources/META-INF/resources/schema.xml [hunk 1]
```diff
@@ -592,7 +772,9 @@
 		<dynamicField docValues="true" indexed="true" multiValued="false" name="*Number_sortable" stored="true" type="double" />
 		<dynamicField docValues="true" indexed="true" multiValued="false" name="*String_sortable" stored="true" type="string" />
 		<dynamicField docValues="true" indexed="true" multiValued="false" name="*_sortable" stored="true" type="string" />
+		<dynamicField indexed="true" multiValued="false" name="*_String_sortable.keyword_lowercase" stored="false" type="string_keyword_lowercase" />
```
Comment (BryanEngler) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/resources/META-INF/resources/schema.xml:775]: i think you can delete this line. *_sortable will also catch *_String_sortable and the mapping is the same

**Comment by BryanEngler** [comment]
looks pretty good. just need to fix all the IndexedFieldsTest's

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/resources/META-INF/resources/schema.xml (hunk 1 above)
Comment (ericyanLr) [reply modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/resources/META-INF/resources/schema.xml:775]: That makes sense to me, I'll try it out. 

As a side note, would that also mean this existing line is also unneccessary?
```
<dynamicField docValues="true" indexed="true" multiValued="false" name="*String_sortable" stored="true" type="string" />
```

**Comment by ericyanLr** [comment]
Made suggested changes and resent for further review: https://github.com/liferay-search/liferay-portal/pull/965

**Pull request closed without merging**